proxy/proxy.go

```diff
@@ -189,11 +191,18 @@ func (p *Proxy) proxyHTTPRequest(
 	httpService ingress.HTTPOriginProxy,
 	isWebsocket bool,
 	disableChunkedEncoding bool,
+	pathRegexp *ingress.Regexp,
+	pathReplacement string,
 	fields logFields,
 ) error {
 	roundTripReq := tr.Request
-	if isWebsocket {
+	if isWebsocket || pathReplacement != "" {
```

Comment:
I'm not sure you need this check here. We still need the isWebsocket, but I don't think we need the pathReplacement check as well.

Reply:
I wasn't sure either, but at least tests pass and everything including logging seem to work both with and without that check.
I removed request cloning and moved rewriting right after logging in ProxyHTTP, like in @regnaio branch.